src/Microsoft.AspNet.Mvc.ViewFeatures/ViewFeatures/AttributeDictionary.cs

```diff
@@ -0,0 +1,653 @@
+            }
+        }
+
+        public void Add(string key, string value)
+        {
+            Add(new KeyValuePair<string, string>(key, value));
+        }
+
+        public bool Contains(KeyValuePair<string, string> item)
+        {
+            if (Count == 0)
+            {
+                return false;
+            }
+
+            var index = Find(item.Key);
+            if (index < 0)
+            {
+                return false;
+            }
+            else
+            {
+                return string.Equals(item.Value, Get(index).Value, StringComparison.OrdinalIgnoreCase);
+            }
+        }
+
+        public bool ContainsKey(string key)
+        {
+            if (Count == 0)
+            {
+                return false;
+            }
+            
+            return Find(key) >= 0;
+        }
+
+        public void CopyTo(KeyValuePair<string, string>[] array, int arrayIndex)
+        {
+            for (var i = 0; i < Count; i++)
+            {
+                array[arrayIndex + i] = Get(i);
+            }
+        }
+
+        public Enumerator GetEnumerator()
+        {
+            return new Enumerator(this);
+        }
+
+        public bool Remove(KeyValuePair<string, string> item)
+        {
+            if (Count == 0)
+            {
+                return false;
+            }
+
+            var index = Find(item.Key);
+            if (index < 0)
+            {
+                return false;
+            }
+            else if (string.Equals(item.Value, Get(index).Value, StringComparison.OrdinalIgnoreCase))
+            {
+                Remove(index);
+                return true;
+            }
+            else
+            {
+                return false;
+            }
+        }
+
+        public bool Remove(string key)
+        {
+            if (Count == 0)
+            {
+                return false;
+            }
+
+            var index = Find(key);
+            if (index < 0)
+            {
+                return false;
+            }
+            else
+            {
+                Remove(index);
+                return true;
+            }
+        }
+
+        public bool TryGetValue(string key, out string value)
+        {
+            if (Count == 0)
+            {
+                value = null;
+                return false;
+            }
+
+            var index = Find(key);
+            if (index < 0)
+            {
+                value = null;
+                return false;
+            }
+            else
+            {
+                value = Get(index).Value;
+                return false;
+            }
+        }
+
+        IEnumerator IEnumerable.GetEnumerator()
+        {
+            return GetEnumerator();
+        }
+
+        IEnumerator<KeyValuePair<string, string>> IEnumerable<KeyValuePair<string, string>>.GetEnumerator()
+        {
+            return GetEnumerator();
+        }
+
+        public struct Enumerator : IEnumerator<KeyValuePair<string, string>>
```

Comment:
This is the standard pattern for Enumerators'n'such - should we open a bug on the BCL?